File: best_time_buy_sell/max_profit.dart
class Solution {
  int maxProfit(List<int> prices) {
    if (prices.length == 1) {
      return 0;
    }
    var maxProfit = 0;
    var minPrice = prices[0];

    for (var i = 1; i < prices.length; i++) {
      if (prices[i] < minPrice) {
        minPrice = prices[i];
      } else {
        if (prices[i] - minPrice > maxProfit) {
          maxProfit = prices[i] - minPrice;
        }
      }
    }
    return maxProfit;
  }
}

File: best_time_buy_sell/max_profit.go
package besttimebuysell

func maxProfit(prices []int) int {
	if len(prices) == 1 {
		return 0
	}

	maxProfit := 0
	minPrice := prices[0]

	for i := 1; i < len(prices); i++ {
		if prices[i] < minPrice {
			minPrice = prices[i]
		} else {
			if prices[i]-minPrice > maxProfit {
				maxProfit = prices[i] - minPrice
			}
		}
	}

	return maxProfit
}
